java-checks/src/main/java/org/sonar/java/checks/VisibleForTestingUsageCheck.java

```diff
@@ -42,7 +42,8 @@
     "com.google.common.annotations.VisibleForTesting",
     "org.assertj.core.util.VisibleForTesting",
     "androidx.annotation.VisibleForTesting",
-    "org.apache.flink.annotation.VisibleForTesting"
+    "org.apache.flink.annotation.VisibleForTesting",
+    "org.sonar.java.annotation.VisibleForTesting"
```

Comment:
I think that's not the best possible fix. This annotation is internal machinery, and it should probably not be included directly in the rule, to not encourage people using OUR implementation (even if the annotation is not described in the rule, some people might look at our open-source code to get the list of supported annotations).

What about another angle, and consider all the annotations called `@VisibleForTested`, whatever their fully qualified name? I think it would be cleaner (but requires another ticket)

Reply:
Created a ticket for it https://jira.sonarsource.com/browse/SONARJAVA-3623